libs/platform/kernel/ebpf_platform_kernel.c

```diff
@@ -761,6 +761,25 @@ ebpf_platform_thread_id()
     return (uint32_t)(uintptr_t)PsGetCurrentThreadId();
 }
 
+_IRQL_requires_max_(PASSIVE_LEVEL) _Must_inspect_result_ ebpf_result_t
+    ebpf_platform_get_authentication_id(_Out_ uint64_t* authentication_id)
+{
+    SECURITY_SUBJECT_CONTEXT context = {0};
+    SeCaptureSubjectContext(&context);
+    LUID local_authentication_id;
+
+    PACCESS_TOKEN access_token = SeQuerySubjectContextToken(&context);
```

Comment:
Was curious why we're not explicitly checking the ```acess_token``` for a non-null value before passing it as argument to the next call to ```SeQueryAuthenticationIdToken```? 

A quick peek at ```SeQueryAuthenticationIdToken``` implementation (minkernel/ntos/se/tokenqry.c) shows that API to assume a non-null parameter value.

Reply:
Per MSDN, `SeQuerySubjectContextToken` is not expected to fail today. If it starts failing sometime in future, I think that will be a regression. I have also added an assert, so we should be able to catch it in our debug tests, if the API behavior changes in a future release.

Reply:
Added checks and also a comment stating these APIs are not expected to fail.

Reply:
Also added traces for failure cases.